File: proxy.js
const express = require('express');
const cors = require('cors');
const axios = require('axios');
const app = express();
const port = 8080;

// 启用 CORS
app.use(cors());
app.use(express.json());

// 静态文件服务
app.use(express.static('./'));

// 代理请求处理
app.post('/proxy', async (req, res) => {
  try {
    const { url, method, headers, body } = req.body;
    const response = await axios({
      method,
      url,
      headers,
      data: body
    });
    res.json(response.data);
  } catch (error) {
    console.error('Proxy Error:', error.response?.data || error.message);
    res.status(500).json({
      error: error.message,
      details: error.response?.data || 'No additional details'
    });
  }
});

app.listen(port, () => {
  console.log(`Proxy server running at http://127.0.0.1:${port}`);
}); 

File: proxy.py
from http.server import HTTPServer, SimpleHTTPRequestHandler
import urllib.request
import json
import ssl

class CORSRequestHandler(SimpleHTTPRequestHandler):
    def end_headers(self):
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, POST, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type, Authorization')
        SimpleHTTPRequestHandler.end_headers(self)

    def do_OPTIONS(self):
        self.send_response(200)
        self.end_headers()

    def do_POST(self):
        if self.path == '/proxy':
            content_length = int(self.headers['Content-Length'])
            post_data = self.rfile.read(content_length)
            data = json.loads(post_data.decode('utf-8'))

            try:
                # 创建请求
                req = urllib.request.Request(
                    data['url'],
                    data=json.dumps(data['body']).encode('utf-8'),
                    headers=data['headers'],
                    method='POST'
                )

                # 发送请求
                with urllib.request.urlopen(req) as response:
                    response_data = response.read()

                # 发送响应
                self.send_response(200)
                self.send_header('Content-type', 'application/json')
                self.end_headers()
                self.wfile.write(response_data)

            except Exception as e:
                self.send_response(500)
                self.send_header('Content-type', 'application/json')
                self.end_headers()
                self.wfile.write(json.dumps({'error': str(e)}).encode('utf-8'))
        else:
            self.send_response(404)
            self.end_headers()

    def do_GET(self):
        return SimpleHTTPRequestHandler.do_GET(self)

def run(server_class=HTTPServer, handler_class=CORSRequestHandler, port=8000):
    server_address = ('', port)
    httpd = server_class(server_address, handler_class)
    print(f'Starting server on port {port}...')
    httpd.serve_forever()

if __name__ == '__main__':
    run() 